### PR TITLE
Add docs for Middleman static site generator

### PR DESCRIPTION
**Pull request by kotovalexarian**

### CODE REVIEW

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ description:
 
 # Static-site generators
 
-Static-site generators like Hugo, VuePress, Nextjs, and Jekyll are all incredibly popular platforms building websites quickly. This guide walks through how to integrate IFPS into each of these workflows.
+Static-site generators like Hugo, VuePress, Nextjs, Middleman, and Jekyll are all incredibly popular platforms building websites quickly. This guide walks through how to integrate IFPS into each of these workflows.
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:8]:
```suggestion
Static-site generators like Hugo, Jekyll, Middleman, Nextjs, and VuePress are all incredibly popular platforms building websites quickly. This guide walks through how to integrate IFPS into each of these workflows.
```

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 2]
```diff
@@ -46,6 +46,46 @@ npx all-relative
 
 Upload the `dist` folder to IPFS.
 
+## Middleman
+
+Refer the Middleman's [Installation](https://middlemanapp.com/basics/install/) guide to install Ruby and Middleman.
+
+Refer the Middleman's [Starting a New Site](https://middlemanapp.com/basics/start-new-site/) guide to set up your project.
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:53]:
```suggestion
```

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 3]
```diff
@@ -46,6 +46,46 @@ npx all-relative
...
+
+In `config.rb` enable relative links:
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:55]:
```suggestion
1. Enable relative links in your project's `config.rb` file:
```

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 4]
```diff
@@ -46,6 +46,46 @@ npx all-relative
...
+
+```ruby
+set :relative_links, true
+```
+
+Links generated by the `link_to` helper or by Markdown markup will become relative, except links to the root.
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:61]:
```suggestion
Links generated by the `link_to` helper or by markdown will become relative except links to the root of your site.
```

Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:61]:
```suggestion
    ```ruby
    set :relative_links, true
    ```

    Links generated by the `link_to` helper or by Markdown markup will become relative, except links to the root.
```

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 5]
```diff
@@ -46,6 +46,46 @@ npx all-relative
...
+
+In `config.rb`, to generate relative links to the root, add the following helper:
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:63]:
```suggestion
1. In `config.rb`, to generate relative links to the root, add the following helper:
```

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 6]
```diff
@@ -46,6 +46,46 @@ npx all-relative
...
+    "#{"../" * (current_page.url.count('/') - 1)}index.html"
+  end
+end
+```
+
+Then use it in templates without the `link_to` helper:
+
+```erb
+<a href="<%= relative_root_path %>">Home</a>
+```
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:77]:
```suggestion
    ```ruby
    helpers do
      def relative_root_path
        "#{"../" * (current_page.url.count('/') - 1)}index.html"
      end
    end
    ```

1. Then use it in templates without the `link_to` helper:

    ```erb
    <a href="<%= relative_root_path %>">Home</a>
    ```
```

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 7]
```diff
@@ -46,6 +46,46 @@ npx all-relative
...
+
+To build a static site:
+
+```bash
+middleman build
+```
+
+Output will be in `./build` by default.
+
+Upload the `build` folder to IPFS.
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:87]:
```suggestion
1. Build your static site:

    ```bash
    middleman build
    ```

    Middleman will output your site to the `./build` folder.

1. Upload the `build` folder to IPFS.
```

**Review by johnnymatthews — APPROVED** [review]
Added some minor formatting changes.

File: docs/how-to/websites-on-ipfs/static-site-generators.md [hunk 8]
```diff
@@ -46,6 +46,45 @@ npx all-relative
...
+Refer the Middleman's [Installation](https://middlemanapp.com/basics/install/) guide to install Ruby and Middleman.
+
+
+1. Enable relative links in your project's `config.rb` file:
+
+```ruby
+set :relative_links, true
+```
+
+Links generated by the `link_to` helper or by markdown will become relative except links to the root of your site.
```
Comment (johnnymatthews) [inline docs/how-to/websites-on-ipfs/static-site-generators.md:60]:
```suggestion
    ```ruby
    set :relative_links, true
    ```

    Links generated by the `link_to` helper or by markdown will become relative except links to the root of your site.
```

**Comment by johnnymatthews** [comment]
This is great! Thanks for adding Middleman to this section.

**Comment by kotovalexarian** [comment]
The list is invalid because all items have number 1.

**Comment by johnnymatthews** [comment]
> The list is invalid because all items have number 1.

Markdown parses recognise ordered lists, and automatically apply the correct number when converting into HTML. Adding `1.` to the start of each line makes it easier to write and edit markdown, since you don't have to re-order the entire list if you were to add another item into the middle of the list.

image

**Pull request merged**